src/configurator/validators/offline.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+const defaultPattern = [ "**/*.*", "!**/*.html", "index.html" ]
+const defaultOfflineConfig = {
+  appcache: true,
+  serviceWorker: true,
+  pattern: defaultPattern,
+}
+
+export default (
+  { config, errors }:
+  { config: StatinamicConfig, errors: Array<string>}
+) => {
+  // Disable offline for development
+  if (config.dev) {
+    config.offline = false
+    return
+  }
+
+  if (!config.offline) {
+    return
+  }
+
+  if (!config.offlineConfig) {
```

Comment:
Do you remember our chat on gitter about `flow` doesn't allow type mutation ? 

for ex:

```js
// define it
config.appcache: Array<string> | false

// Function
writeAppcache(pattern: Array<string>)

// Now call it

if (
    config.appcache &&
    typeof config.appcache !== "boolean"
  ) {
    writeAppcache(config.appcache)
}
```